src/test/store/profile-view.test.js

```diff
@@ -835,6 +835,148 @@ describe('actions/ProfileView', function() {
       expect(getMarker(markerIndexes[0]).name.includes('a')).toBeTruthy();
       expect(getMarker(markerIndexes[1]).name.includes('b')).toBeTruthy();
     });
+
+    it('filters the markers by a potential data payload of type FileIO', function() {
+      const profile = getProfileWithMarkers([
+        ['a', 0, null],
+        ['b', 1, null],
+        ['c', 2, null],
+        [
+          'd',
+          3,
+          {
+            cause: { stack: 2, time: 1 },
+            endTime: 1024,
+            filename: '/foo/bar/',
+            operation: 'create/open',
+            source: 'PoisionOIInterposer',
+            startTime: 1022,
+            type: 'FileIO',
+          },
+        ],
+      ]);
+      const { dispatch, getState } = storeWithProfile(profile);
+
+      expect(
+        selectedThreadSelectors.getSearchFilteredMarkerIndexes(getState())
+      ).toHaveLength(4);
+      dispatch(ProfileView.changeMarkersSearchString('/foo/bar/'));
```

Comment:
Ideally we should check various things here:
1. we should check if this works for partial text too (for example searching for just "foo")
2. we need to test if this filters the operation and the source too (so you need to change the search string again below, and test again).